test/maven.collector.test.ts

```diff
@@ -0,0 +1,60 @@
+import { expect } from 'chai';
+import { PomXmlDependencyCollector } from '../src/collector';
+
+describe('Maven pom.xml parser test', () => {
+    const collector:PomXmlDependencyCollector = new PomXmlDependencyCollector();
+
+    it('tests valid pom.xml', async () => {
+        const deps = await collector.collect(
+        `<dependencyManagement>
+            <dependencies>
+                <!-- Dependency with scope as runtime -->
+                <dependency>
+                    <groupId>{a.groupId}</groupId>
+                    <artifactId>bc</artifactId>
+                    <version>{a.version}</version>
+                </dependency>
+
+                <!-- Dependency with scope as runtime -->
+                <dependency>
+                    <groupId>b</groupId>
+                    <artifactId>c-d</artifactId>
+                    <version>1.2.3</version>
+                    <scope>runtime</scope>
+                </dependency>
+
+                <!-- Dependency with scope as compile -->
+                <dependency>
+                    <groupId>c</groupId>
+                    <artifactId>ab-cd</artifactId>
+                    <version>2.3</version>
+                    <scope>compile</scope>
+                    <optional>true</optional>
+                </dependency>
+
+                <!-- Dependency with scope as test -->
+                <dependency>
+                    <groupId>d</groupId>
+                    <artifactId>AB-CD</artifactId>
+                    <version>3.4.5.6</version>
+                    <scope>test</scope>
+                    <optional>true</optional>
+                </dependency>
+            </dependencies>
+         </dependencyManagement>
+        `);
+        expect(deps.length).equal(3);
+        expect(deps[0]).is.eql({
+          name: {value: '{a.groupId}:bc', position: {line: 0, column: 0}},
+          version: {value: '{a.version}', position: {line: 7, column: 30}}
+        });
+        expect(deps[1]).is.eql({
+          name: {value: 'b:c-d', position: {line: 0, column: 0}},
+          version: {value: '1.2.3', position: {line: 14, column: 30}}
+        });
+        expect(deps[2]).is.eql({
+          name: {value: 'c:ab-cd', position: {line: 0, column: 0}},
+          version: {value: '2.3', position: {line: 22, column: 30}}
+        });
+    });
+});
```

Comment:
@dvandra Could you please add few more negative tests as well?
like 
1.  empty string
2. empty dependencyManagement
3. empty dependencyManagement + dependencies
4. without any score
5. with only test scope

Reply:
Ack!